Makefile.dep

```diff
@@ -116,3 +116,9 @@ ifneq (,$(filter test_utils_interactive_sync,$(USEMODULE)))
     DEFAULT_MODULE += test_utils_interactive_sync_shell
   endif
 endif
+
+# always register a peripheral driver as a required feature when the corresponding
+# module is requested
+ifneq (,$(filter periph_%,$(USEMODULE)))
+  FEATURES_REQUIRED += $(filter-out periph_common,$(filter periph_%,$(USEMODULE)))
```

Comment:
There relation between `periph_%` modules and `periph_%` is not only for `periph_common` different. At least `periph_init_%` modules also have no corresponding feature. There may be more. `makefiles/features_modules.inc.mk` does contain a bit of insight.

It is unlikely, though, that this would cause any practical issues. I think the conversion from features to modules is done after the dependency resolution and users typically never explicitly ask for initialization, but instead rather have to disable initialization to not get them.

Reply:
I moved the check into the mentioned Makefile since the list of `periph` modules not having a corresponding feature defined seems to be already present there.